public/network-realms.mjs

```diff
@@ -497,6 +497,10 @@ class VirtualPlayer extends HeadTrackedEntity {
   unlink(realm) {
     this.cleanupMapFns.get(realm)();
     this.cleanupMapFns.delete(realm);
+
+    if (!this.headTracker.isLinked()) {
+      this.dispatchEvent(new MessageEvent('leave'));

```

Comment:
This isn't isomorphic. This class should not emit `leave` if it does not emit entry.

Further, this code should be in the cleanup function.

Reply:
Addressed in https://github.com/webaverse/multiplayer-do/pull/6